taskmgr/queueservice.go

```diff
@@ -186,16 +189,16 @@ func (q queueService) deleteFromAzure(taskId string, deleteUrl string) error {
 }
 
 // Retrieves the number of tasks requested from the Azure queues.
-func (q *queueService) pollTaskUrl(taskQueue *TaskQueue, ntasks int, logger *logrus.Entry) ([]*TaskRun, error) {
+func (q *queueService) pollTaskUrl(taskQueue *taskQueue, ntasks int) ([]*TaskRun, error) {
 	taskRuns := []*TaskRun{}
-	queueMessagesList := &QueueMessagesList{}
+	var r queueMessagesList
 	// To poll an Azure Queue the worker must do a `GET` request to the
 	// `signedPollUrl` from the object, representing the Azure queue. To
 	// receive multiple messages at once the parameter `&numofmessages=N`
 	// may be appended to `signedPollUrl`. The parameter `N` is the
 	// maximum number of messages desired, `N` can be up to 32.
 	// Since we can only process one task at a time, grab only one.
-	resp, _, err := httpbackoff.Get(fmt.Sprintf("%s%s%d", taskQueue.SignedPollUrl, "&numofmessages=", ntasks))
+	resp, _, err := httpbackoff.Get(fmt.Sprintf("%s%s%d", taskQueue.SignedPollUrl, "&numofmessages=", int(math.Min(32, float64(ntasks)))))
```

Comment:
I'm so stupid - of course Min is correct - my mistake! I should only review code in the afternoons!!

Isn't `len(taskRuns)` always zero, because it is instantiated as an empty slice of `*TaskRun` just before this is evaluated?